### DISCUSSION TITLE
devbuild fails: ClrHost.obj : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM64'

### DISCUSSION THREAD

**Discussion by joeloff** [wixtoolset/issues#8563] [WiX Development]
I tried running `devbuild.cmd` after forking the repository. I double checked and have the latest VCS bits (x86/x64/arm64), all the .NET framework targeting and developer packs, NET 6, 7, 8 and 9. I'm on VS 2022 (17.9.0). I have all the required workloads listed installed as well.

![image](https://github.com/wixtoolset/issues/assets/7409981/1b3c33b8-f676-4ebb-bfec-1dd186d581e8)

Does this error look familiar. I tried searching this discussion for the message (both open and closed items), but didn't find anything. I'm guessing it's something environmental on my setup.

**Comment by robmen** [comment]
I've never seen that error message before.

**Comment by bevanweiss** [comment]
It looks like you've configured something around the ClrHost.obj such that it has generated it as an x86 type object file.  But then you're trying to link it all into an ARM64 assembly.

I'd suggest that you delete ALL of the files.
And then start with a fresh clone of the repository (AFTER you've forked it if you intend on making changes).

You should also indicate what build environment you're running under.  I assume you're running under an x64 type machine... but assumptions aren't a good idea.  Perhaps you're on an ARM64 Mac trying to do weird things with containers/Parallels...

**Comment by joeloff** [comment]
Found the issue, I selected the arm toolset instead of arm64 😿 

![image](https://github.com/wixtoolset/issues/assets/7409981/58e5527e-ec85-4204-8ffe-595cfe830a01)

I'm going to optimistically close this
